### PR TITLE
ICP-4820 - WWST-653 - Fix activity polling and device Watch

### PR DESCRIPTION
**Pull request by juano2310**

### CODE REVIEW

File: smartapps/smartthings/logitech-harmony-connect.src/logitech-harmony-connect.groovy [hunk 1]
```diff
@@ -528,21 +528,26 @@ def pollResponse(response, data) {
...
 	        ResponseValues.hubs.each {
 		        // Device-Watch relies on the Logitech Harmony Cloud to get the Device state.
 		        def isAlive = it.value.status
-		        def d = getChildDevice("harmony-${it.key}")
-		        d?.sendEvent(name: "DeviceWatch-DeviceStatus", value: isAlive!=504? "online":"offline", displayed: false, isStateChange: true)
-		        if (it.value.message == "OK") {
-					map["${it.key}"] = "${it.value.response.data.currentAvActivity},${it.value.response.data.activityStatus}"
+		        activities.each { activity ->
+					if ("${activity.deviceNetworkId}".contains("harmony-${it.key}")) {
+						def childDevice = getChildDevice(activity.deviceNetworkId)
```
Comment (marstorp) [inline smartapps/smartthings/logitech-harmony-connect.src/logitech-harmony-connect.groovy]: if activities = getChildDevices(), isn't the activity in this each loop the child for which you getChidlDevice(activity.deviceNetworkId)? Do you really need to getChildDevice again here, can't you just use activity, granted the name is a but obscure...?

File: smartapps/smartthings/logitech-harmony-connect.src/logitech-harmony-connect.groovy [hunk 2]
```diff
@@ -528,21 +528,26 @@ def pollResponse(response, data) {
...
+	        def activities = getChildDevices()
 	        ResponseValues.hubs.each {
 		        // Device-Watch relies on the Logitech Harmony Cloud to get the Device state.
 		        def isAlive = it.value.status
-		        def d = getChildDevice("harmony-${it.key}")
-		        d?.sendEvent(name: "DeviceWatch-DeviceStatus", value: isAlive!=504? "online":"offline", displayed: false, isStateChange: true)
-		        if (it.value.message == "OK") {
-					map["${it.key}"] = "${it.value.response.data.currentAvActivity},${it.value.response.data.activityStatus}"
+		        activities.each { activity ->
+					if ("${activity.deviceNetworkId}".contains("harmony-${it.key}")) {
```
Comment (marstorp) [inline smartapps/smartthings/logitech-harmony-connect.src/logitech-harmony-connect.groovy]: indentation.....

File: smartapps/smartthings/logitech-harmony-connect.src/logitech-harmony-connect.groovy [hunk 3]
```diff
@@ -556,31 +561,23 @@ def pollResponse(response, data) {
...
-	                        if (aux2[1] == "1")
-	                            runIn(5, "poll", [overwrite: true])
-	                    } else {
-	                        childDevice?.sendEvent(name: "switch", value: "off")
-	                        if (aux2[1] == "3")
-	                            runIn(5, "poll", [overwrite: true])
-	                    }
-	                }
-	            }
+                def childDevice = getChildDevice(activity.deviceNetworkId)
```
Comment (marstorp) [inline smartapps/smartthings/logitech-harmony-connect.src/logitech-harmony-connect.groovy]: Similare comment as above, isn't activities a list of childDevices? so you shouldn't have to do another getChildDevice() on a specific deviceNetworkId here...

**Review by marstorp — CHANGES_REQUESTED** [review]
Please answer the questions and/or update the code

**Review by surfous — CHANGES_REQUESTED** [review]
The changes themselves look alright, though I have the same questions about using the activity device at hand that Ingvar has asked, rather than re-querying by device ID. 

Indentation is very inconsistent throughout the file, and makes reading the code and identifying blocks very challenging. Please take the time to normalize the indentation throughout for the sake of maintainability. Whomever will work on this in the future (including yourself) will thank you!

**Review by marstorp — APPROVED** [review]
LGTM ⛵️

**Review by surfous — APPROVED** [review]
Thanks for the cleanup - it's much easier to follow the nested blocks of code at a glance.

**Pull request merged**